File: python/Tree.py
class TreeNode:
  def __init__(self, data):
    self.data = data
    self.left = None
    self.right = None

class Tree:
  def __init__(self, root):
    self.root = root

  def preorder(self, root, arr):
    if root == None: return

    arr.append(root.data)

    self.preorder(root.left, arr)
    self.preorder(root.right, arr)

  def inorder(self, root, arr):
    if root == None: return

    self.inorder(root.left, arr)

    arr.append(root.data)

    self.inorder(root.right, arr)

  def postorder(self, root, arr):
    if root == None: return

    self.postorder(root.left, arr)
    self.postorder(root.right, arr)

    arr.append(root.data)

  def levelorder(self, root, arr):
    queue = [root]

    while len(queue) > 0:
      node = queue.pop(0)

      arr.append(node.data)

      if node.left != None:
        queue.append(node.left)

      if node.right != None:
        queue.append(node.right)

  def height(self, root):
    if root == None:
      return 0
    else:
      left = self.height(root.left)
      right = self.height(root.right)

      if left > right:
        return left + 1
      else:
        return right + 1

root = TreeNode(1)
tree = Tree(root)

root.left = TreeNode(2)
root.right = TreeNode(3)

root.left.left = TreeNode(4)
root.left.right = TreeNode(5)

root.right.right = TreeNode(6)

preorder = []
tree.preorder(root, preorder)

print(f"preorder: {preorder}")

inorder = []
tree.inorder(root, inorder)

print(f"inorder: {inorder}")

postorder = []
tree.postorder(root, postorder)

print(f"postorder: {postorder}")

levelorder = []
tree.levelorder(root, levelorder)

print(f"level order: {levelorder}")

print(f"height: {tree.height(root)}")

File: c#/Tree.cs
var root = new TreeNode(1);
var tree = new Tree(root);

root.left = new TreeNode(2);
root.right = new TreeNode(3);

root.left.left = new TreeNode(4);
root.left.right = new TreeNode(5);

root.right.right = new TreeNode(6);

List<int> preorder = new List<int>();

tree.preorder(root, preorder);
tree.print_tree(preorder, "preorder");

List<int> inorder = new List<int>();

tree.inorder(root, inorder);
tree.print_tree(inorder, "inorder");

List<int> postorder = new List<int>();

tree.postorder(root, postorder);
tree.print_tree(postorder, "postorder");

List<int> levelorder = new List<int>();

tree.levelorder(root, levelorder);
tree.print_tree(levelorder, "level order");

Console.WriteLine($"height: {tree.height(root)}");
class TreeNode
{
  public int data;
  public TreeNode left;
  public TreeNode right;

  public TreeNode(int data)
  {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}

class Tree {
  public TreeNode root;

  public Tree(TreeNode root)
  {
    this.root = root;
  }

  public void print_tree(List<int> tree, string text)
  {
    Console.Write($"{text}: [");

    for (int i = 0; i < tree.Count; i++) {
      if (i == tree.Count - 1) {
        Console.Write($"{tree[i]}]\n");
      } else {
        Console.Write($"{tree[i]}, ");
      }
    }
  }

  public void preorder(TreeNode root, List<int> arr)
  {
    if (root == null) return;

    arr.Add(root.data);

    this.preorder(root.left, arr);
    this.preorder(root.right, arr);
  }

  public void inorder(TreeNode root, List<int> arr)
  {
    if (root == null) return;

    this.inorder(root.left, arr);

    arr.Add(root.data);

    this.inorder(root.right, arr);
  }

  public void postorder(TreeNode root, List<int> arr)
  {
    if (root == null) return;

    this.postorder(root.left, arr);
    this.postorder(root.right, arr);

    arr.Add(root.data);
  }

  public void levelorder(TreeNode root, List<int> arr)
  {
    List<TreeNode> queue = new List<TreeNode>();

    queue.Add(root);

    while (queue.Count > 0) {
      TreeNode node = queue[0];

      queue.RemoveAt(0);

      arr.Add(node.data);

      if (node.left != null) {
        queue.Add(node.left);
      }

      if (node.right != null) {
        queue.Add(node.right);
      }
    }
  }

  public int height(TreeNode root)
  {
    if (root == null) {
      return 0;
    } else {
      int left = this.height(root.left);
      int right = this.height(root.right);

      if (left > right) {
        return left + 1;
      } else {
        return right + 1;
      }
    }
  }
}